pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java

```diff
@@ -2241,7 +2468,7 @@ public long getLong(@Positive int columnIndex) throws SQLException {
     // hold strong references to user objects (e.g. classes -> classloaders), thus it might lead to
     // OutOfMemory conditions.
     @Override
-    public synchronized Throwable fillInStackTrace() {
```

Comment:
This synchronized is interesting. 

IMO the synchronized here was unnecessary as `return this` is atomic. Personally I'm comfortable removing the synchronized and not including a `lock() / unlock()` block around the `return this`.

Reply:
It certainly does not need to use a `PgResultSet` instance's `lock`. But it is also not using a `PgResultSet` instance to synchronize on now (as a static instance).

Presumably this is synchronized because the method on `Throwable` being overridden is synchronized.
The parent method is synchronized because it (potentially) mutates member variables. This clearly is not, so removing synchronization is fine.